Assets/Voxelgon/Math/Position.cs

```diff
@@ -1,43 +1,67 @@
 using UnityEngine;
 
 namespace Voxelgon.Math {
-	public struct Position {
-		short x;
-		short y;
-		short z;
-
-		public Position(short x, short y, short z) {
-			this.x = x;
-			this.y = y;
-			this.z = z;
-		}
-
-		public Position(float x, float y, float z) {
-			this.x = (short) x;
-			this.y = (short) y;
-			this.z = (short) z;
-		}
-
-		public Position(Vector3 v) {
-			this.x = (short) v.x;
-			this.y = (short) v.y;
-			this.z = (short) v.z;
-		}
-
-		public static explicit operator Position(Vector3 v) {
-			return new Position(v);
-		}
-
-		public static explicit operator Vector3(Position pos) {
-			return new Vector3(pos.x, pos.y, pos.z);
-		}
-
-		public static bool operator ==(Position a, Position b) {
-			return (a.x == b.x && a.y == b.y && a.z == b.z);
-		}
-
-		public static bool operator !=(Position a, Position b) {
-			return (a.x != b.x || a.y != b.y || a.z != b.z);
-		}
-	}
+    public struct Position {
+        
+        public bool Equals(Position other)
+        {
+            return _x == other._x && _y == other._y && _z == other._z;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            return obj is Position && Equals((Position) obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = _x.GetHashCode();
+                hashCode = (hashCode*397) ^ _y.GetHashCode();
+                hashCode = (hashCode*397) ^ _z.GetHashCode();
+                return hashCode;
+            }
+        }
+
+        readonly short _x;
```

Comment:
Move variables above methods, see [this](http://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods/310967#310967).